core/include/detray/core/detail/container_views.hpp

```diff
@@ -142,6 +140,18 @@ typename T::const_view_type get_data(const T& viewable) {
 template <typename T>
 using dvector_view = vecmem::data::vector_view<T>;
 
+/// Get the vecmem view type of a vector
+template <typename T, typename A>
+dvector_view<T> get_data(std::vector<T, A>& vec) {
+    return vecmem::get_data(vec);
+}
+
+/// Get the vecmem view type of a vector - const
+template <typename T, typename A>
+dvector_view<const T> get_data(const std::vector<T, A>& vec) {
+    return vecmem::get_data(vec);
+}
+
```

Comment:
In the buffer header, I used a custom ```get_data``` function for the buffers that returns a view by value instead of by reference (```vecmem::get_data(vector_buffer)``` returns a reference to ```vector_view```), so that the ```dmulti_view``` types get generated from the buffers as exactly the same types as they are declared in the ```view_type``` of the original class they belong to. So the issue was ```dmulti_view<..., dmulti_view<dvector_view<>, ...>, ...>``` vs ```dmulti_view<..., dmulti_view<dvector_view<>&, ...>, ...>```. Therefore, in the view header I unfortunately also could not pick up the ```vecmem::get_data``` functions by ```namespace detray{ using vecmem::get_data }``` anymore, because that would result in a conflicting declaration with the functions I had to add for the buffers, so I had to pull in ```get_data``` for the vector views by hand, too